src/PrestaShopBundle/Install/Install.php

```diff
@@ -227,13 +241,26 @@ public function generateSettingsFile(
      * Replace "parameters.yml" with "parameters.php" in "app/config".
      *
      * @param array $parameters
-     *
-     * @return bool|int
+     * @param bool $useEnvValuesIfExists
+     * @return bool
      */
-    public function processParameters($parameters)
+    public function processParameters(array $parameters, bool $useEnvValuesIfExists = false): bool
     {
-        $parametersContent = sprintf('<?php return %s;', var_export($parameters, true));
-        if (!file_put_contents(_PS_ROOT_DIR_ . '/app/config/parameters.php', $parametersContent)) {
+        $content = 'array(' . PHP_EOL;
+        $content .= chr(9) . "'parameters' => array(" . PHP_EOL;
+
+        foreach ($parameters['parameters'] as $parameterKey => $parameterValue) {
```

Comment:
I think we can unit test this logic 😉  . You can look at my PR https://github.com/PrestaShop/PrestaShop/pull/25919 where I test functions that output PHP sample files.

Reply:
Moreover we need to encapsulate this logic into a function or a class. This script is already too long.